corehq/apps/data_dictionary/views.py

```diff
@@ -143,6 +147,117 @@ def data_dictionary_json(request, domain, case_type_name=None):
+            Prefetch('allowed_values', queryset=CasePropertyAllowedValue.objects.order_by('allowed_value'))
+        )
+
+        current_url = request.build_absolute_uri()
+        links = {"self": update_url_query_params(current_url, {"skip": skip, "limit": limit})}
+        if skip:
+            links["previous"] = update_url_query_params(
+                current_url,
+                {"skip": max(skip - limit, 0), "limit": limit}
+            )
+        if case_type_data["properties_count"] > (skip + limit):
+            links["next"] = update_url_query_params(current_url, {"skip": skip + limit, "limit": limit})
+        case_type_data["_links"] = links
+
+        case_type_data["groups"] = []
+        data_validation_enabled = toggles.CASE_IMPORT_DATA_DICTIONARY_VALIDATION.enabled(domain)
+        used_props = used_props_by_case_type.get(case_type.name, [])
+        for group_id, props in itertools.groupby(properties_queryset, key=attrgetter("group_id")):
+            props = list(props)
+            grouped_properties = []
+            for prop in props:
+                prop_data = {
+                    'id': prop.id,
+                    'description': prop.description,
+                    'label': prop.label,
+                    'fhir_resource_prop_path': fhir_resource_prop_by_case_prop.get(prop),
+                    'name': prop.name,
+                    'deprecated': prop.deprecated,
+                    'is_safe_to_delete': prop.name not in used_props and prop.name != geo_case_prop,
+                    'index': prop.index,
+                }
+                if data_validation_enabled:
+                    prop_data.update({
+                        'data_type': prop.data_type,
+                        'allowed_values': {
+                            av.allowed_value: av.description
+                            for av in prop.allowed_values.all()
+                        },
+                    })
+                grouped_properties.append(prop_data)
+
+            group_data = {
+                "name": "",
+                "properties": grouped_properties
+            }
+            # Note that properties can be without group
+            if group_id:
+                group = props[0].group
+                group_data.update({
+                    "id": group.id,
+                    "name": group.name,
+                    "description": group.description,
+                    "deprecated": group.deprecated,
+                })
+            case_type_data["groups"].append(group_data)
+
+        return JsonResponse(case_type_data)
+    else:
+        queryset = CaseType.objects.filter(domain=domain).annotate(properties_count=Count('property'))
+        if not request.GET.get("load_deprecated_case_types", False) == "true":
+            queryset = queryset.filter(is_deprecated=False)
+        case_types_data = [_get_case_data(case_type) for case_type in queryset]
```

Comment:
Do we need the full case type JSON when just retrieving the list of case types and their property counts? I imagine properties like `is_safe_to_delete` and `module_count` we only need when loading the full set of data for a case type (since this is when a user will be able to delete the case type).

Reply:
Great point. I went with the expected response in the tech spec but I agree on this. I can check on this and get rid of things that we do not need.

Reply:
After thinking/looking more on this, I think logically all the case type related data including `is_safe_to_delete` and  `module_count` should be part of the Case Type API. 
Instead, we should consider not returning these case type details when we use the case properties endpoint - just return case type name and their grouped properties.
Does that makes sense ? @zandre-eng @kaapstorm

Reply:
I'm okay with what you've done in f91eaf5ee39cd1c9a657af3a9f20df63ad4a6112 -- I'm feeling quite pragmatic about this. The viewmodel is the only thing that will be consuming this API, and as long as it has what it needs before it needs it, I think that works.